src/lib/y2network/hwinfo.rb

```diff
@@ -17,14 +17,69 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
+require "yast"
+
 module Y2Network
-  # Physical interface hardware information
+  # Stores useful (from networking POV) items of hwinfo for an interface
+  # FIXME: decide whether it should read hwinfo (on demand or at once) for a network
+  # device and store only necessary info or just parse provided hash
   class Hwinfo
-    # @return [Hash] hwinfo
-    attr_accessor :data
+    attr_reader :hwinfo
+
+    def initialize(name:)
+      # FIXME: store only what's needed.
+      @hwinfo = load_hwinfo(name)
+    end
+
+    # Shortcuts for accessing hwinfo items. Each hwinfo item has own method for reading
+    # its value. There are two exceptions however. First exception is hwinfo["name"] item
+    # which carries device model human friendly description. This item is accessible via
+    # {Hwinfo::description}. Second exception is hwinfo["dev_name"] item which can be read
+    # through {Hwinfo::dev_name} or its alias {Hwinfo::name}.
+    [
+      { name: "dev_name", default: "" },
+      { name: "mac", default: "" },
+      { name: "busid", default: "" },
+      { name: "link", default: false },
+      { name: "driver", default: "" },
+      { name: "drivers", default: [] },
+      { name: "requires", default: [] },
+      { name: "hotplug", default: false },
+      { name: "wl_auth_modes", default: "" },
+      { name: "wl_enc_modes", default: nil },
+      { name: "wl_channels", default: nil },
+      { name: "wl_bitrates", default: nil }
+    ].each do |hwinfo_item|
+      method_name = hwinfo_item[:name].downcase
+
+      # @!method #{method_name} read value of #{hwinfo_item[:name]} from hwinfo or
```

Comment:
Unfortunately, YARD does not execute the code, and the reader will be just confused

Compare, `to_integer` is rendered as
https://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast/Convert#to_integer-class_method
and documented as
https://github.com/yast/yast-ruby-bindings/blob/4288ac455902ac43aeec0175523657a6a538e251/src/ruby/yast/convert.rb#L19